File: project-euler/euler_dart/test/problem_003_test.dart
import 'package:test/test.dart';

import 'package:euler_dart/problem_003.dart';

void main() {
  test("largestPrimeFactor(2) should return 2", () {
    expect(largestPrimeFactor(2), 2);
  });
  test("largestPrimeFactor(5) should return 5", () {
    expect(largestPrimeFactor(5), 5);
  });
  test("largestPrimeFactor(7) should return 7", () {
    expect(largestPrimeFactor(7), 7);
  });
  test("largestPrimeFactor(13195) should return 29", () {
    expect(largestPrimeFactor(13195), 29);
  });
  test("largestPrimeFactor(600851475143) should return 6857", () {
    expect(largestPrimeFactor(600851475143), 6857);
  });
}

File: project-euler/euler-java/src/test/java/com/github/adrianomoreira/projecteuler100/Problem003Test.java
package com.github.adrianomoreira.projecteuler100;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

public class Problem003Test {

    @DisplayName("largestPrimeFactor(2) should return 2")
    @Test
    public void test(){
        Assertions.assertEquals(2, Problem003.largestPrimeFactor(2));
    }

    @DisplayName("largestPrimeFactor(3) should return 3")
    @Test
    public void test2(){
        Assertions.assertEquals(3, Problem003.largestPrimeFactor(3));
    }
    @DisplayName("largestPrimeFactor(5) should return 5")
    @Test
    public void test3(){
        Assertions.assertEquals(5, Problem003.largestPrimeFactor(5));
    }
    @DisplayName("largestPrimeFactor(7) should return 7")
    @Test
    public void test4(){
        Assertions.assertEquals(7, Problem003.largestPrimeFactor(7));
    }
    @DisplayName("largestPrimeFactor(29) should return 13195")
    @Test
    public void test5(){
        Assertions.assertEquals(29, Problem003.largestPrimeFactor(13195));
    }
    @DisplayName("largestPrimeFactor(6857) should return 600851475143")
    @Test
    public void test6(){
        Assertions.assertEquals(6857, Problem003.largestPrimeFactor(600851475143L));
    }

}
